src/AST/Translator.Expressions.dfy

```diff
@@ -413,9 +417,8 @@ module Bootstrap.AST.Translator.Expressions {
     decreases ASTHeight(le), 0
   {
     var lhss := ListUtils.ToSeq(le.LHSs);
-    var bvs :- Seq.MapResult(lhss, (pat: C.CasePattern<C.BoundVar>) reads * =>
-      :- Need(pat.Var != null, Invalid("Null pattern variable in let expression."));
-      Success(TypeConv.AsString(pat.Var.Name)));
+    var bvs := Seq.Map((pat: C.CasePattern<C.BoundVar>) reads * =>
+      if pat.Var == null then "_" else TypeConv.AsString(pat.Var.Name), lhss);
```

Comment:
Is it ever going to be an issue that we assign the `_` variable?

Reply:
I'm honestly not sure what to do here. It does come up in practice that `pat.Var == null`, and I'm not sure how to interpret it. Do you have any ideas about what the semantics of that should be?

Reply:
I think `pat = null` is indeed the `_` case. I'm guessing it's safe to translate this to a var called `_`, but let's add a TODO here for when we support better pattern matching?

Reply:
I refactored it so that it could produce `Unsupported` in the case where a bound variable is null.